File: snake-strategy-bono/src/main/java/co/electric/snake/strategy/bonostrategy/newdirectionprocessor/ByFilteredDirections.java
package co.electric.snake.strategy.bonostrategy.newdirectionprocessor;

import co.electric.snake.framework.model.Direction;
import co.electric.snake.strategy.bonostrategy.BlockingDirectionContainer;
import co.electric.snake.strategy.bonostrategy.SimpleDirectionContainer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ByFilteredDirections implements NewDirectionProcessor {

    private static final int ORDER = 4;

    private static final Logger LOG = LoggerFactory.getLogger(ByFilteredDirections.class);

    @Override
    public Direction process(SimpleDirectionContainer filteredDirections, SimpleDirectionContainer equivalentBestDirections, BlockingDirectionContainer blockingDirections) {
        Direction newDirection = null;

        if (!filteredDirections.isEmpty()) {
            newDirection = processFinalDirection(filteredDirections, LOG);
        }

        return newDirection;
    }

    @Override
    public int getOrder() {
        return ORDER;
    }

}

File: snake-strategy-bono/src/main/kotlin/co/electric/snake/strategy/bonostrategy/newdirectionprocessor/ByFilteredDirections.kt
package co.electric.snake.strategy.bonostrategy.newdirectionprocessor

import co.electric.snake.framework.model.Direction
import co.electric.snake.strategy.bonostrategy.BlockingDirectionContainer
import co.electric.snake.strategy.bonostrategy.SimpleDirectionContainer
import org.slf4j.LoggerFactory
import java.util.*

class ByFilteredDirections : NewDirectionProcessor {

    companion object {
        private val LOG = LoggerFactory.getLogger(ByFilteredDirections::class.java)
    }

    override val order = 5

    override fun process(filteredDirections: SimpleDirectionContainer, equivalentBestDirections: SimpleDirectionContainer, blockingDirections: BlockingDirectionContainer): Optional<Direction> {
        return processFinalDirection(filteredDirections, LOG)
    }

}
